setup_version.py

```diff
@@ -0,0 +1,348 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, immuneML Development Team.
+# Distributed under the LGPLv3 License. See LICENSE for more info.
+"""
+immuneML.
+
+Copyright (c) 2020, immuneML Development team.
+
+This file generates the version info and appends a ".beta"
+to the logo if needed.
+
+"""
+import argparse
+import json
+import os
+import pathlib
+import subprocess
+
+
+# Define path to files we are interacting with:
+# 1) The file containing the current version info:
+CURRENT_VERSION_FILE = pathlib.Path('version.json')
+# 2) The file with version info accessible by the immuneML library:
+VERSION_FILE = pathlib.Path('source').joinpath('version.py')
+# 3) The setup.py file:
+SETUP_PY = pathlib.Path('setup.py')
+# 4) The info.py file containing the logo:
+INFO_FILE = pathlib.Path('source').joinpath('info.py')
+
+# Define formats for released and development versions:
+VERSION_DEV_FMT = '{major:d}.{minor:d}.{micro:d}.dev{dev:d}'
+VERSION_FMT = '{major:d}.{minor:d}.{micro:d}'
+
+# Define the text for creating .py files for immuneML:
+HEADER_TXT = """# -*- coding: utf-8 -*-
+# Copyright (c) 2020, immuneML Development Team.
+# Distributed under the LGPLv3 License. See LICENSE for more info."""
+
+VERSION_TXT = '''{header:s}
+"""Version information for immuneML.
+
+This file is generated by immuneML (``setup_version.py``)
+"""
+SHORT_VERSION = '{version:s}'
+VERSION = '{version:s}'
+FULL_VERSION = '{full_version:s}'
+GIT_REVISION = '{git_revision:s}'
+GIT_VERSION = '{git_version:s}'
+RELEASE = {release:}
+
+if not RELEASE:
+    VERSION = GIT_VERSION
+
+MIN_PYTHON_VERSION = {MIN_PYTHON_VERSION}
+'''
+
+INFO_TXT = '''{header:s}
+"""This module just contains some info for immuneML.
+
+Here we define the name of the program and some other
+relevant info.
+
+This file is generated by immuneML (``setup_version.py``)
+"""
+PROGRAM_NAME = 'immuneML'
+URL = 'https://greifflab.org/software/'
+GIT_URL = 'https://github.com/uio-bmi/ImmuneML'
+CITE = """
+[1] x. yyy, z. kkk, J. Journaly McJounralFace, 2021
+    doi: https://soon.666
+
+"""
+LOGO = r"""{logo:s}"""
+'''
+
+
+# Define the immuneML logo:
+LOGO = r"""
+                                                ____   ____  ___
+   ____  __  __    __  __   __  __ __  _______  \   \ /   /  \ /
+   \  / /  |/  |  /  |/  | / / / /   |/ / ___/  | |\ \ /| |  | |
+   / / / / | / | / / | / |/ / / / /| | / _/     | | \_/ | |  | |
+  / / / /|  /| |/ /|  /| | /_/ / / |  / /___    | |     | |  | |___
+ /__\/_/ |_/ |_|_/ |_/ |_|____/_/  |_/______\  /___\   /___\ /_____\
+
```

Comment:
nice :)
if at all possible, it would be cool to have an antibody-M like in our logo: https://drive.google.com/drive/folders/1j1VdNfVxYllVJ9LfvVOdj3tvihMKw0aa